File: ports/java/src/core/global/BcsCoreTypes.java
package bcs.core.global;

public class BcsCoreTypes {
    public enum BcsGlobalColor {
        WHITE,
        BLACK,
        RED,
        GREEN,
        BLUE,
        TRANSPARENT
    }

    public static void warning(String msg) {
        System.out.println("[BCS Warning] " + msg);
    }

    public static void critical(String msg) {
        System.out.println("[BCS Critical] " + msg);
    }
}
    // BcsAlignment mapping (using int constants since Java Enums don't support bitwise easily)
    public static final int ALIGN_LEFT = 0x0001;
    public static final int ALIGN_RIGHT = 0x0002;
    public static final int ALIGN_HCENTER = 0x0004;
    public static final int ALIGN_JUSTIFY = 0x0008;
    public static final int ALIGN_TOP = 0x0020;
    public static final int ALIGN_BOTTOM = 0x0040;
    public static final int ALIGN_VCENTER = 0x0080;
    public static final int ALIGN_CENTER = ALIGN_VCENTER | ALIGN_HCENTER;

    public enum BcsOrientation {
        HORIZONTAL(1),
        VERTICAL(2);

        public final int value;
        private BcsOrientation(int value) {
            this.value = value;
        }
    }
}

File: ports/csharp/src/core/global/BcsCoreTypes.cs
using System;

namespace BCS.Core.Global {
    public enum BcsGlobalColor {
        White,
        Black,
        Red,
        Green,
        Blue,
        Transparent
    }

    public static class BcsCoreLogging {
        public static void Warning(string msg) {
            Console.WriteLine($"[BCS Warning] {msg}");
        }

        public static void Critical(string msg) {
            Console.WriteLine($"[BCS Critical] {msg}");
        }
    }
}

namespace BCS.Core.Global {
    [System.Flags]
    public enum BcsAlignment {
        AlignLeft = 0x0001,
        AlignRight = 0x0002,
        AlignHCenter = 0x0004,
        AlignJustify = 0x0008,
        AlignTop = 0x0020,
        AlignBottom = 0x0040,
        AlignVCenter = 0x0080,
        AlignCenter = AlignVCenter | AlignHCenter
    }

    public enum BcsOrientation {
        Horizontal = 0x1,
        Vertical = 0x2
    }
}
